### ISSUE TITLE
Implement email notifications

### ISSUE THREAD

**Issue by gehinz** [agdsn/pycroft#267]
Adding email notifications that remind members to pay at the end of a month if not enough money is left, would help with user satisfaction and lower the support demand. The notification should also mention that if a member wants to stop his membership an email is sufficient for resigning.
Email notifications for the following situations would help too:

- welcome message to detect typos in mailforwardings
- confirmation that the membership was canceled
- Notification if the network access gets blocked (including the reason). An opt-out should be possible
- low traffic credit warning (adjustable via our website and with a default of 1d*80%)

**Comment by toothstone** [comment]
This is a partial duplicate of #23 and blocked by #258. From past experience, I don't think manually handling bounces due to typos in the addresses is feasible.

**Comment by lukasjuhrich** [comment]
@Paktosan since I doubt you're currently working on this due to your amount of other, more important projects, I'm going to remove you as an assignee.

**Comment by lukasjuhrich** [comment]
Which of these notifications are we missing? I found the following usages of relevant methods:
- `task.mail_negative_members` which does what it says
- `task.execute_scheduled_tasks` which sends a mail if the task failed.

**Comment by ibot3** [comment]
These notifications are currently missing:

- [ ] confirmation that the membership was canceled
- [ ] Notification if the network access gets blocked (including the reason). 
- [ ] confirmation that a transaction has been matched automatically (good) or manually (bad, remind to use the correct payment description)
- [ ] confirmation that a task has been scheduled

Also the mail that notified about a negative balance should also include the current payment details and the personalized payment description.

**Comment by LukasGibeh** [comment]
@ibot3 @gehinz 
I would propose another notification to remind members when their rental contract with Studentenwerk is running out that their AG DSN membership is not ending automatically.
